haystack/nodes/label_generator/pseudo_label_generator.py

```diff
@@ -0,0 +1,211 @@
+
+        :param question_producer: The question producer used to generate questions or a list of already produced
+        questions/document pairs in Dict format {"question": "question text ...", "document": "document text ..."}.
+        :type question_producer: Union[QuestionGenerator, List[Dict[str, str]]]
+        :param retriever: The retriever used to query document stores
+        :type retriever: BaseRetriever
+        :param cross_encoder_model_name_or_path: The path to the cross encoder model, defaults to
+        cross-encoder/ms-marco-MiniLM-L-6-v2
+        :type cross_encoder_model_name_or_path: str (optional)
+        :param max_questions_per_document: The max number of questions generated per document, defaults to 3
+        :type max_questions_per_document: int
+        :param top_k: The number of answers retrieved for each question, defaults to 50
+        :type top_k: int (optional)
+        """
+
+        super().__init__()
+        self.question_document_pairs = None
+        self.question_generator = None
+        if isinstance(question_producer, QuestionGenerator):
+            self.question_generator = question_producer
+        elif isinstance(question_producer, list) and len(question_producer) > 0:
+            example = question_producer[0]
+            if isinstance(example, dict) and "question" in example and "document" in example:
+                self.question_document_pairs = question_producer
+            else:
+                raise ValueError("question_producer list must contain dicts with keys 'question' and 'document'")
+        else:
+            raise ValueError("Provide either a QuestionGenerator or nonempty list of questions/document pairs")
+
+        self.retriever = retriever
+        self.cross_encoder = CrossEncoder(cross_encoder_model_name_or_path)
+        self.max_questions_per_document = max_questions_per_document
+        self.top_k = top_k
+
+    def generate_questions(self, documents: List[Document]) -> List[Dict[str, str]]:
+        """
+        It takes a list of documents and generates a list of question-document pairs
+
+        :param documents: A list of documents to generate questions from
+        :type documents: List[Document]
+        :return: A list of question-document pairs.
+        """
+        question_doc_pairs: List[Dict[str, str]] = []
+        if self.question_document_pairs:
+            question_doc_pairs = self.question_document_pairs
+        else:
+            questions = self.question_generator.generate_batch([d.content for d in documents])
+            for idx, question_list_per_doc in enumerate(questions):
+                for q in question_list_per_doc[: self.max_questions_per_document]:
+                    question_doc_pairs.append({"question": q.strip(), "document": documents[idx].content})
+        return question_doc_pairs
+
+    def mine_negatives(self, question_doc_pairs: List[Dict[str, str]], batch_size: int = 2) -> List[Dict[str, str]]:
```

Comment:
Why is the `batch_size` here set to 2? I'd make it optional (default None) and use the value set in the init if no batch_size is set here.